modules/http-client/src/main/java/com/devonfw/module/httpclient/common/impl/AbstractAsyncServiceHttpClient.java

```diff
@@ -117,4 +117,15 @@ protected Object handleUnsupportedBody(Object body) {
    */
   protected abstract HttpRequest createRequest(ServiceClientInvocation<S> invocation);
 
+  // START
+  @Override
+  protected <R> CompletableFuture<R> getCompletableFuture(ServiceClientInvocation<S> invocation) {
+
+    long startTime = System.nanoTime();
+    HttpRequest request = createRequest(invocation);
+    return (CompletableFuture<R>) this.client.getHttpClient().sendAsync(request, BodyHandlers.ofString())
+        .thenApply(HttpResponse::body);
```

Comment:
When adding new features to devon4j we should always consider testing and documenting them.
You can not cast from `CompletableFuture<String>` to `CompletableFuture<R>` so IMHO this code can not work properly (except `R` happens to be `String`). There is the existing logic already present from the other `call` methods that  transforms the `HttpResponse` to the result (`R`) by unmarshalling from JSON. You should reuse this and transform the `CompletableFuture` based on that logic.